test/cartesian.jl

```diff
@@ -159,6 +167,25 @@ end
             @test all(i->A[i]==A[R[i]], R)
             @test all(i->A[i]==A[R[i]], collect(R))
             @test all(i->i in R, collect(R))
+
+            # Indexing a CartesianIndices with another CartesianIndices having the same ndims
+            # forwards the indexing to the component ranges and retains the wrapper
+            @test R[R] === R
+
+            R2 = R[(1:1 for _ in 1:length(oinds))...]
+            @test R2 isa CartesianIndices{ndims(R)}
+            @test R2[(1 for _ in 1:length(oinds))...] == R[(1 for _ in 1:length(oinds))...]
```

Comment:
Maybe test against ordinary arrays, i.e. `R2 = collect(R)[(1:1 for _ in 1:length(oinds))...]` ?